build/pxc-entrypoint.sh

```diff
@@ -483,4 +484,53 @@ if [ "$1" = 'mysqld' -a -z "$wantHelp" ]; then
 	fi
 fi
 
+function get_primary() {
+    peer-list -on-start=/usr/bin/get-pxc-state -service=$PXC_SERVICE 2>&1 \
+        | grep wsrep_ready:ON:wsrep_connected:ON:wsrep_local_state_comment:Synced:wsrep_cluster_status:Primary \
+        | sort \
+        | tail -1 || true
+}
+
+IS_PRIMARY_EXISTS=$(get_primary)
+if [ -f ${DATADIR}/grastate.dat ]; then
+    NOT_SAVE_BOOTSTRAP=$(cat ${DATADIR}/grastate.dat | grep 'safe_to_bootstrap: 0' || true)
+fi
+
+function node_recovery() {
+    echo "Recovery is in progress, please wait...."
+    sed -i 's/safe_to_bootstrap: 0/safe_to_bootstrap: 1/g' ${DATADIR}/grastate.dat
+    sed -i 's/wsrep_cluster_address=.*/wsrep_cluster_address=gcomm:\/\//g' /etc/mysql/node.cnf
+    rm -f /tmp/recovery-case
+    exec mysqld
+}
+
+if [[ -z "$IS_PRIMARY_EXISTS" && -n "$NOT_SAVE_BOOTSTRAP" ]] || [[ -z "$IS_PRIMARY_EXISTS" && -f "${DATADIR}/gvwstate.dat" ]]; then
+    trap node_recovery USR1
+    touch /tmp/recovery-case
+
+    if [[ -z ${seqno} ]] ||  [[ ${seqno} == '-1' ]]; then
+        seqno=$(mysqld --wsrep_recover 2>&1 | grep 'Recovered position' | awk -F':' '{print $NF}' || true)
```

Comment:
can we have the same parsing as in lines 467-471 ?

Reply:
it works for all possible versions of PXC